docs/design/generics/details.md

```diff
@@ -2505,8 +2505,24 @@ binary operator:
 -   _Same-type constraints_: `where`...`==`...
 -   _Implements constraints_: `where`...`impls`...
 
+And there are two positions that `where` can be written:
+
+-   Attached to an `impl as` declaration, before the body of the impl. Here it
+    is used to assign a value to an
+    [associated constant](#associated-constants).
+    ```
+    impl Class as Interface where .A = i32 { ... }
+    ```
+-   Inside a type expression. Here it is used to constrain the possible values
+    of facets and/or their associated constants.
+    ```
+    fn F[T: Interface where T.A impls OtherInterface](t: T) { ... }
```

Comment:
We currently don't allow you to use the name of the binding inside the type of the binding, so this is written instead like:

```suggestion
    fn F[T: Interface where .A impls OtherInterface](t: T) { ... }
```